File: NCTUcourses/PL/hw1/sol/hw1.pl
sub fib

{
  
   if ($h{"fib($_[0])"}==undef) { 
      if ($_[0]<=1) { $h{"fib($_[0])"} = $_[0]; }
      else { $h{"fib($_[0])"} = fib($_[0]-1)+fib($_[0]-2); }
   } 
   return $h{"fib($_[0])"};
}

print "Enter an integer>=0: ";
$n=<stdin>*1;

print "The $n","th Fibonacci number is ",fib($n),".\n";
print "Below is the hash table created during the computation.\n";
while (($key,$value)=each %h) { 
   
   print "$key => $value\n"; 

}

File: NCTUcourses/PL/hw1/sol/hw1.f
      program McCarthy
      implicit none
      integer :: n,f
      print *, "Enter an integer >=0"
      read *, n
      print *, "f(", n,") =",f(n)
      end program McCarthy

      function f(n)
      implicit none
      integer :: n,f,fs
      fs=1
      do while (fs>0)
         if (n>100) then
             n=n-10
             fs=fs-1
         else
             n=n+11
             fs=fs+1
         end if
      end do
      f=n
      end function f

